File: satoken/src/main/kotlin/com/satoken/config/ApiEnumParameterCustomizer.java
package com.satoken.config;


import com.satoken.enums.BasicEnum;
import com.satoken.enums.BasicEnumCustomizer;
import io.swagger.v3.oas.models.media.Schema;
import io.swagger.v3.oas.models.parameters.Parameter;
import org.springdoc.core.customizers.ParameterCustomizer;
import org.springframework.core.MethodParameter;
import org.springframework.stereotype.Component;

/**
 * 枚举参数自定义配置
 *
 * @author vains
 */
@Component
public class ApiEnumParameterCustomizer implements ParameterCustomizer, BasicEnumCustomizer {

    @Override
    public Parameter customize(Parameter parameterModel, MethodParameter methodParameter) {
        Class<?> parameterType = methodParameter.getParameterType();

        // 枚举处理
        if (BasicEnum.class.isAssignableFrom(parameterType)) {
            if(parameterModel==null) {
               parameterModel =  new Parameter();
            }
            parameterModel.setDescription(getDescription(parameterType));

            Schema<Object> schema = new Schema<>();
            schema.setEnum(getValues(parameterType));
            parameterModel.setSchema(schema);
        }

        return parameterModel;
    }
}

File: satoken/src/main/kotlin/com/satoken/enums/ApiEnumParameterCustomizer.kt
package com.satoken.enums

import io.swagger.v3.oas.models.media.Schema
import io.swagger.v3.oas.models.parameters.Parameter
import org.springdoc.core.customizers.ParameterCustomizer
import org.springframework.core.MethodParameter
import org.springframework.stereotype.Component

/**
 * 枚举参数自定义配置
 */
@Component
class ApiEnumParameterCustomizer : ParameterCustomizer, BasicEnumCustomizer {

    override fun customize(parameterModel: Parameter?, methodParameter: MethodParameter): Parameter? {
        val parameterType = methodParameter.parameterType
        if (parameterModel != null) {
            // 枚举处理
            if (BasicEnum::class.java.isAssignableFrom(parameterType)) {
                parameterModel.description = getDescription(parameterType)
                val schema = Schema<Any>()
                schema.enum = getValues(parameterType)
                parameterModel.schema = schema
            }
        }
        return parameterModel
    }
}
